### PR TITLE
Task solved

### PR DESCRIPTION
**Pull request by AriohDK**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_SIZE = 10;
+    private K[] keyArray;
+    private V[] valueArray;
+    private int storegeSize = 0;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private int size;
```
You do not need to add 'storage' to var names. We are in Storage class, so, all variables belongs to Storage

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,51 @@
...
+                storegeSize++;
+                break;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        int index = 0;
+        for (K keys : keyArray) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:40]: use fori loop and remove this 'index' var

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,51 @@
...
+            index++;
+            if ((keys != null && keys.equals((key)) || (keys == null && keys == key))) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:42]: `keys == null && keys == key`
**simplify it**

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,51 @@
...
+    private V[] valueArray;
+    private int storegeSize = 0;
+
+    public StorageImpl() {
+        this.keyArray = (K[]) new Object[ARRAY_SIZE];
+        this.valueArray = (V[]) new Object[ARRAY_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:17]: This method is not correct. We have poor test cases... Try to test it:
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_SIZE = 10;
+    private K[] key;
+    private V[] value;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]: plural
image

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,17 +3,48 @@
...
+
+    public StorageImpl() {
+        this.key = (K[]) new Object[ARRAY_SIZE];
+        this.value = (V[]) new Object[ARRAY_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (isEquals(this.key[i], key)) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:19]:
```suggestion
            if (isEquals(keys[i], key)) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -3,17 +3,48 @@
...
+                this.value[i] = value;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:20]:
```suggestion
                values[i] = value;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -3,17 +3,48 @@
...
+                return;
+            }
+        }
+        this.key[size] = key;
+        this.value[size] = value;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:25]:
```suggestion
        keys[size] = key;
        values[size] = value;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 9]
```diff
@@ -3,17 +3,48 @@
...
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < this.key.length; i++) {
+            if (isEquals(this.key[i], key)) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:32]:
```suggestion
            if (isEquals(keys[i], key)) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 10]
```diff
@@ -3,17 +3,48 @@
...
+            }
+        }
+        this.key[size] = key;
+        this.value[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < this.key.length; i++) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:31]:
```suggestion
        for (int i = 0; i < keys.length; i++) {
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 11]
```diff
@@ -3,17 +3,48 @@
...
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private boolean isEquals(K firstKey, K secondKey) {
+        if (firstKey == secondKey || (firstKey != null && firstKey.equals(secondKey))) {
+            return true;
+        }
+        return false;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:48]:
```suggestion
        return firstKey == secondKey || (firstKey != null && firstKey.equals(secondKey));
```

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**